CRISTAL-165: Initialize the main layout

tkrieck opened the pull request:
# Jira URL

https://jira.xwiki.org/browse/CRISTAL-165

# Changes

* Provides some updates to Vuetify styles and components
* Changed the default page width to fill all space available
* Updated the temporary DS menu

## Description

This PR is mostly about updating Vuetify to make it more inline with the Shoelace implementation. Also other changes like:
- Full width page: to gather feedback on the default state of the page this version release the horizontal space limitation
- The DS menu was changed to make it visually more clear of all the different environments that are available.

## Clarifications

* Most components still need reworks to make them generic
* DSFR was not touched at all and should not be used

# Screenshots & Video

Vuetify Implementation

<img width="1544" alt="Screenshot 2024-04-23 at 15 19 03" src="https://github.com/xwiki-contrib/cristal/assets/149672322/dcc9f711-d9a1-409a-b3ac-0ba45b8fb2a6">


Shoelace implementation:

<img width="1541" alt="Screenshot 2024-04-23 at 15 18 22" src="https://github.com/xwiki-contrib/cristal/assets/149672322/ef692dd3-8c8a-4951-8f85-bd54e3aebb95">

# Executed Tests

-

# Expected merging strategy

* Prefers squash: Yes <!-- No — Explain why. -->
* Backport on branches:
  * N/A

manuelleduc commented on ds/shoelace/src/vue/css/style.css:
```diff
@@ -6,19 +6,16 @@
   
-    /* Fonts */
-    --cr-base-font-size: 16px;
+    /* Fonts */    
```
```suggestion
    /* Fonts */
```

manuelleduc commented:
I'm not sure to like the result on large screen (here about 2000px wide).
![image](https://github.com/xwiki-contrib/cristal/assets/327856/5be54820-0d2c-46e7-a6e2-73046a40949a)

The like doc info actions and doc page actions feels misplaced in the middle of nowhere.
I'd suggest going through a forum proposal to discuss the finer details of content width as we know not everybody agrees of the correction solution.

manuelleduc commented on lib/src/index.css:
```diff
@@ -55,15 +55,20 @@ html {
-
+/* 
 .content-scroll > div {
   width: 100%;
-  max-width: var(--cr-sizes-max-page-width); /*TODO: This value needs to be dynamic*/
+  max-width: var(--cr-sizes-max-page-width);
+} */
```
dead code

manuelleduc commented on lib/src/index.css:
```diff
@@ -55,15 +55,20 @@ html {
+} */
+
+.center-content{
+  width: 100%;
```
Is this 100% the same as `--cr-sizes-max-page-width`?

tkrieck replied:
No, the logic here is that content will span 100% until met the max-width set on `--cr-sizes-max-page-width`. Now, depending on user preference both CAN be the same, when the user wants the content to span the entire length of the viewport.

`--cr-sizes-max-page-width` will need to be dynamic, but it's fixed until we have this feature ready.

manuelleduc commented on skin/src/images/no-one.jpg:
Could it be a svg to make it lighter?

tkrieck replied:
Probably, I'll check for an SVG version. I did it with a jpg for now because user photos will usually be in this format.

manuelleduc commented on ds/shoelace/src/vue/css/style.css:
```diff
@@ -6,19 +6,16 @@
-
-   --cr-sizes-max-page-width: 960px;
+   --cr-sizes-max-page-width: 100%;
    --cr-sizes-left-sidebar-width: 250px;
    --cr-sizes-collapsed-sidebar-width: 32px;
-
-
+   --cr-base-font-size: 16px;
+   
     /*
      * Typography
```
I missed that earlier, but the whole file deserves a good formatting pass.
Note to myself, I need to check how to lint css files for correct formatting too.s

tkrieck replied:
I am using Prettier and Stylelint on VS Codium and its not returning any formatting errors. The base file was adapted from Shoelace's. https://github.com/shoelace-style/shoelace/blob/next/src/themes/light.css

manuelleduc replied:
Yes, I've noticed we are missing something on eslint.
A few notes:
- eslint has since a few weeks a major version 9 change, and the new configuration format is already lightly documents, making the migration quite challenging (I did a first quick failed attempt yesterday morning)
- we need to decide for which editors we optimize: I'd say IntelliJ and VSCode

manuelleduc commented on ds/vuetify/src/components/vuetifyDesignSystemLoader.ts:
```diff
@@ -71,16 +71,16 @@ export class VuetifyDesignSystemLoader implements DesignSystemLoader {
     });
     app.use(vuetify);
     // Native Vuetify components
-    app.component("XAvatar", VAvatar);
```
`VAvatar` and `VImg` are now unused imports [above](https://github.com/xwiki-contrib/cristal/pull/181/files#diff-3e699a289e0d9f5de5e6e3a4d6e51893653f36c9b9694b5c6f941446850a64e4L28-L35)

@tkrieck let me know if you need help setting up an IDE, that usually helping to automatically fix this kind of issues.

tkrieck replied:
I am using VSCodium, and indeed it is warning me about those two. In the rush to get the PR going I did not see them. My bad.

manuelleduc commented on ds/vuetify/src/vue/x-avatar.vue:
```diff
@@ -0,0 +1,11 @@
+  props: {
+    image: { type: String, required: true },
+    size: { type: String, required: true },
+},
```
```suggestion
  },
```

manuelleduc commented on ds/vuetify/src/vue/x-img.vue:
```diff
@@ -0,0 +1,10 @@
+  props: { src: { type: String, required: true } },
+};
+</script>
+  
```
```suggestion

```

manuelleduc commented on skin/src/vue/c-config.vue:
```diff
@@ -71,3 +48,67 @@ export default {
   },
 };
 </script>
+<template>
```
Definitely much nicer now, thanks!

manuelleduc commented on skin/src/vue/c-config.vue:
```diff
@@ -71,3 +48,67 @@ export default {
+    <div class="grid-container">
+      <div v-for="wikiConfig in configList" :key="wikiConfig.name">
+          <div>
+          <div class="wiki-name">{{ wikiConfig.name }}
+            <span class="offline" v-if="wikiConfig.offline">Offline</span>
+          </div>
```
```suggestion
          <div class="wiki-name">
            {{ wikiConfig.name }}
            <span class="offline" v-if="wikiConfig.offline">Offline</span>
          </div>
```

manuelleduc commented on skin/src/vue/c-help.vue:
```diff
@@ -3,8 +3,7 @@ import { CIcon } from "@cristal/icons";
   <div class="help-button">
-    <x-btn pill size="small"
-      ><c-icon name="question-circle"></c-icon> XWiki Help</x-btn
+    <x-btn pill size="small"><c-icon name="question-circle"></c-icon> XWiki Help</x-btn
```
```suggestion
    <x-btn pill size="small">
      <c-icon name="question-circle"></c-icon>
      XWiki Help
    </x-btn>
```

manuelleduc commented:
Also, we currently have a CI issue, but running `pnpm run lint` returns the following results:

```
 $ pnpm run lint

> @0.2.0 lint /home/mleduc/cristal/cristal
> nx run-many -t lint


   ✔  nx run @cristal/icons:lint (2s)
   ✔  nx run @cristal/api:lint (2s)

———————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————
   ✖  nx run @cristal/skin:lint
      > @cristal/skin@1.0.0 lint /home/mleduc/cristal/cristal/skin
      > eslint "./src/**/*.{ts,tsx,vue}" --max-warnings=0
      
      
      /home/mleduc/cristal/cristal/skin/src/vue/c-config.vue
         55:9   error    Delete `··`                                prettier/prettier
         56:34  error    Insert `⏎············`                     prettier/prettier
         57:35  warning  Attribute "v-if" should go before "class"  vue/attributes-order
         59:32  warning  Attribute "v-if" should go before "class"  vue/attributes-order
         60:74  error    Insert `⏎··········`                       prettier/prettier
         63:33  warning  Attribute "v-if" should go before "class"  vue/attributes-order
        114:1   error    Delete `⏎`                                 prettier/prettier
      
      /home/mleduc/cristal/cristal/skin/src/vue/c-help.vue
        6:29  error  Insert `⏎······`  prettier/prettier
      
      ✖ 8 problems (5 errors, 3 warnings)
        5 errors and 3 warnings potentially fixable with the `--fix` option.
```

tkrieck commented:
> I'm not sure to like the result on large screen (here about 2000px wide). ![image](https://private-user-images.githubusercontent.com/327856/325129972-5be54820-0d2c-46e7-a6e2-73046a40949a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM5NTE5NzcsIm5iZiI6MTcxMzk1MTY3NywicGF0aCI6Ii8zMjc4NTYvMzI1MTI5OTcyLTViZTU0ODIwLTBkMmMtNDZlNy1hNmUyLTczMDQ2YTQwOTQ5YS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDI0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQyNFQwOTQxMTdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02MGJkMmNmYzBlNjNjNzhiNzFlNGI0YTk2YzgxOTBhZTdjZTM2ZjhjMWVlNmJiYzE4NzdiOGQ5NTcyMTU5MjQwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.A8ryuJHpElW2l0ZgNZQiKAJ0taVF8IZ325X7pq1GD8o)
> 
> The like doc info actions and doc page actions feels misplaced in the middle of nowhere. I'd suggest going through a forum proposal to discuss the finer details of content width as we know not everybody agrees of the correction solution.

Yeah, that's not ideal. Your screenshot also has an bug, the title should be aligned with the content. I'll start a forum discussion.

tkrieck commented:
> Also, we currently have a CI issue, but running `pnpm run lint` returns the following results:
> 
> ```
>  $ pnpm run lint
> 
> > @0.2.0 lint /home/mleduc/cristal/cristal
> > nx run-many -t lint
> 
> 
>    ✔  nx run @cristal/icons:lint (2s)
>    ✔  nx run @cristal/api:lint (2s)
> 
> ———————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————
>    ✖  nx run @cristal/skin:lint
>       > @cristal/skin@1.0.0 lint /home/mleduc/cristal/cristal/skin
>       > eslint "./src/**/*.{ts,tsx,vue}" --max-warnings=0
>       
>       
>       /home/mleduc/cristal/cristal/skin/src/vue/c-config.vue
>          55:9   error    Delete `··`                                prettier/prettier
>          56:34  error    Insert `⏎············`                     prettier/prettier
>          57:35  warning  Attribute "v-if" should go before "class"  vue/attributes-order
>          59:32  warning  Attribute "v-if" should go before "class"  vue/attributes-order
>          60:74  error    Insert `⏎··········`                       prettier/prettier
>          63:33  warning  Attribute "v-if" should go before "class"  vue/attributes-order
>         114:1   error    Delete `⏎`                                 prettier/prettier
>       
>       /home/mleduc/cristal/cristal/skin/src/vue/c-help.vue
>         6:29  error  Insert `⏎······`  prettier/prettier
>       
>       ✖ 8 problems (5 errors, 3 warnings)
>         5 errors and 3 warnings potentially fixable with the `--fix` option.
> ```

I've run through all errors thrown by lint, it should be ok now. There were also some references to non existent files when I converted images to svg, these were also fixed.

manuelleduc commented:
> I've run through all errors thrown by lint, it should be ok now. There were also some references to non existent files when I converted images to svg, these were also fixed.

The CI is now working :)
You can get the results here https://ci.xwiki.org/job/XWiki%20Contrib/job/cristal/job/vuetify-updates/

The most recent build has unit tests failing, but I believe it's because you are not fully up to date with the main branch.

manuelleduc commented:
@tkrieck small issue (probably a missing revert): on edit mode, the content is aligned left instead of being centered.

tkrieck commented:
> @tkrieck small issue (probably a missing revert): on edit mode, the content is aligned left instead of being centered.

Yep, it was a property that I missed. Fixed now.
